parsers/mcmaster-parser-2.js

```diff
@@ -102,9 +102,16 @@ function parse(text){
     return timetable;
 }
 
-fs.readFile(filename, 'utf8', function(err, data) {
-    if (err) throw err;
-    console.log(ics.buildICS(parse(data)));
-});
+var convertToCal = function(rawTimetable, directory, filename, callback){
```

Comment:
We're just passing back the filename back to whoever calls convertToCal, since we no longer have intense parsing, we just call ics.buildICS(parse(rawTimetable)) and pipe that to a file.